src/libcore/mem.rs

```diff
@@ -415,6 +415,66 @@ pub fn replace<T>(dest: &mut T, mut src: T) -> T {
     src
 }
 
+/// A guarding type which will abort upon drop.
+///
+/// This is used for catching unwinding and transforming it into abort.
+struct ExitGuard;
+
+impl Drop for ExitGuard {
+    fn drop(&mut self) {
+        // To avoid unwinding, we abort the program, which ensures that the destructor of the
+        // invalidated value isn't runned.
+        unsafe { intrinsics::abort(); }
```

Comment:
Aha. Well, perhaps libstd could do so instead of providing a straight re-export.